src/backend/optimizer/path/allpaths.c

```diff
@@ -1702,7 +1702,10 @@ set_subquery_pathlist(PlannerInfo *root, RelOptInfo *rel,
 		 */
 		if ((!bms_is_empty(required_outer)) &&
 			check_query_for_possible_motion(subquery))
+		{
 			config->force_singleQE = true;
+			config->is_under_subplan = true;
```

Comment:
Although this seems to be a proper change, this line makes the comment above uncomplete and unclear. Could you update it as well?

Also, meaning of the field 'is_under_subplan' has changed - not any subplan results in this flag, and it's used like additional condition to disable min/max optimization, mark config as a correlated subplan and enable direct dispatch (whatever it means).
I suggest to rename this field to reflect it's actual meaning. And provide actual comment near is_under_subplan declaration in header.

Reply:
It seems like `is_under_subplan` is only set [here](https://github.com/arenadata/gpdb/blob/c5d46dd4bb9a0601da22e3c89e9fb09e01414d74/src/backend/optimizer/plan/subselect.c#L638) and in this patch, so it is only set for subplans that originate from `SubLink`s (maybe originally that was the only way to get a subplan?) and now for subplans in lateral joins. Naming it `is_under_sublink_subplan_or_lateral_join` seems wrong, maybe it would be better to split it into two flags, `is_under_sublink_subplan` and `is_under_lateral_join`? Naming it after the way it is used (only in that single condition and only if subplan happens to be correlated) also seems like a bad idea.

Reply:
why it's bad to name things for their purpose?
Also, force_singleQE field is set in this place only, other places are only reading it. Why to set different flag here?

Reply:
Well, it's a really complicated name... Should I name it that?
Yes, but it is used in a completely different place (and `force_singleQE` does reflect its purpose, since it forces the subplan to be executed on a single QE). I could instead rename it to `is_under_lateral_join` and add it as an additional part of the condition, but the condition is already complicated enough. Do you think this would be clearer?

Reply:
I suggest to rename is_under_subplan to something like can_have_dependencies and describe in more details in comments what does it mean now and when set.